File: smashpen/src/main/java/com/smashpen/smashpen/repository/CharacterNotesRepository.java
package com.smashpen.smashpen.repository;

import com.smashpen.smashpen.domain.User;
import com.smashpen.smashpen.domain.SmashCharacter;
import com.smashpen.smashpen.domain.CharacterNotes;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.Collection;

public interface CharacterNotesRepository extends JpaRepository<CharacterNotes, Long> {
	Collection<CharacterNotes> findAllByUserId(Long id);
	CharacterNotes findCharacterNotesBySmashCharacterAndUser(SmashCharacter smashCharacter, User user);
}

File: smashpen/src/main/java/com/smashpen/smashpen/repository/CharacterNotesRepository.kt
package com.smashpen.smashpen.repository

import com.smashpen.smashpen.domain.User
import com.smashpen.smashpen.domain.SmashCharacter
import com.smashpen.smashpen.domain.CharacterNotes
import org.springframework.data.jpa.repository.JpaRepository

interface CharacterNotesRepository : JpaRepository<CharacterNotes, Long> {
    fun findAllByUserId(id: Long?): List<CharacterNotes>
    fun findBySmashCharacterAndUser(smashCharacter: SmashCharacter, user: User): CharacterNotes?
}